en/docs/learn/x509certificate-authenticator.md

```diff
@@ -0,0 +1,703 @@
+
+    ``` toml
+    [authentication.authenticator.x509_certificate.parameters]
+    setClaimURI = "http://wso2.org/claims/identity/userCertificate"
+    ```
+
+4.  To enable storing the X509 certificate as a user claim, add the
+    following parameter.
+
+    ``` toml 
+    [authentication.authenticator.x509_certificate.parameters]
+    EnforceSelfRegistration = true
+    ```
+
+## Adding a claim mapping for the certificate
+
+If storing the certificate as a user claim is enabled, the X509
+certificate will be stored as a user claim and verified with the
+retrieved certificate from the request.
+
+1.  Sign in to the WSO2 IS Management Console with one of the following
+    URLs using `           admin          ` as the **username** and
+    **password**.
+
+    ``` java
+    For HTTP  --> http://<HTTP_HOST>:9776/carbon
+    For HTTPS --> https://<HTTPS_HOST>:9443/carbon
+    ```
+
+2.  On the **Main** tab, click **Claims &gt; Add**.  
+    ![](../assets/img/50501577/103328153.png)
+3.  Click **Add Local Claim**.  
+    ![](../assets/img/50501577/103328154.png)
+4.  Add a new claim for the **certificate** by giving the details as
+    below, e.g., select a mapped attribute for the claim that is
+    supported by the underlying database type.
+    ![](../assets/img/50501577/103328155.png)
+5.  Click **Add**.
+
+## Updating the column size of the database for X509 certificates
+
+Make note of the following points and configure your database to match
+your use case:  
+  
+
+-   [Disabling Certificate Validation in an Unstarted WSO2 IS
+    Pack](#ConfiguringX509CertificateAuthenticator-DisablingCertificateValidationinanUnstartedWSO2ISPack)
+-   [Disabling Certificate Validation in an Already-started WSO2 IS
+    Pack](#ConfiguringX509CertificateAuthenticator-DisablingCertificateValidationinanAlready-startedWSO2ISPack)
+-   [Using an identity claim for the X509 certificate or working with
+    read only user
+    stores](#ConfiguringX509CertificateAuthenticator-UsinganidentityclaimfortheX509certificateorworkingwithreadonlyuserstores)
+-   [Using a wso2 claim for the X509
+    certificate](#ConfiguringX509CertificateAuthenticator-Usingawso2claimfortheX509certificate)
+
+### Using an identity claim for the X509 certificate or working with read only user stores
+
+If you are using an identity claim to store X509 Certificates, e.g.,
+`                   http://wso2.org/claims/identity                  ,        `
+or if you are working with a read-only user store, the certificate gets
+stored in the `         DATA_VALUE        ` column of the
+`         IDN_IDENTITY_USER_DATA        ` table. The default DB script
+sets the column size to 255 characters but in this case, the certificate
+value has more than 255 characters. Therefore, you need to update the
+column size to a higher value.
+
+Follow the steps given below to update the column size:
+
+You do not need to update the column size if you are using WSO2 IS 5.4.0
+or above 5.4.0.
```

Comment:
According to this statement, for 5.10 we don't need to follow these steps so shall we remove this entire section under the "Using an identity claim for the X509 certificate or working with read only user stores" heading?